### DISCUSSION TITLE
Postgres encrypted connection

### DISCUSSION THREAD

**Discussion by mkv27** [supabase/supabase#330] [Questions] [answered]
I tried an encrypted connection through 5432 port. But it does not work.
Awesome project! Thanks!

**Comment by soedirgo** [comment, marked as answer]
Hey @mkv27! We only support password encryption at the moment (CMIIW), so it should be enough to use the connection string in Settings > Database with psql or any other client.

https://www.postgresql.org/docs/current/encryption-options.html

We're working on adding SSL encryption support, so watch this space!

**Reply by thepeoplesbourgeois** [reply]
The password is sent via basic-auth as part of the URL, though... C M I **_I_** W, but unless that request is being sent over an encrypted channel, can't anybody with access to your network traffic see that password, plain as day?

If so, this looks like a great opportunity for some PGP (Prolific Grabbing of Passwords) [EDIT: added this line out of the exasperation upon discovering "CMIIW" was _not_, in fact, an encryption protocol.]

**Reply by inian** [reply]
This has now been fixed for new projects. You can find the SSL cert in Database and PgBouncer Settings page.

**Comment by thepeoplesbourgeois** [comment]
Postgres usernames and passwords are only used as an authorization mechanism for performing actions on a particular database. They have no part in handling encryption, either of the contents of that database, or any connections to it, be they local or remote.

You should probably enable SSL or GSSAPI..... fairly soon. Or months ago.
